cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -1163,6 +1175,118 @@ protected Future<Void> clusterId(KafkaStatus kafkaStatus) {
                 });
     }
 
+    public Future<Void> canScaleDownBrokers(KafkaStatus kafkaStatus) {
```

Comment:
I think all these methods should have its own home in some separate class. We want to keep the classes small and focused on partitcular thing. That makes it easier to update them, test them etc.